### PR TITLE
refactor: add view updates for account SLEs

### PR DESCRIPTION
**Pull request by mvadari**
## High Level Overview of Change

This PR adds a line to `SetAccount` and `SetRegularKey`:
```cpp
ctx_.view().update(sle);
```
This commits the changes of the account SLE to the ledger.

### Context of Change

The changes to the account SLE are committed in transaction processing anyways, so this isn't technically a bug and doesn't change anything about how transactions are processed. However, it does make the code more readable and understandable, especially since `AccountSet` and `SetRegularKey` are very simple transactions that devs new to rippled might be looking at to better understand how to write a new transaction.

### Type of Change

- [x] Refactor (non-breaking change that only restructures code)
- [x] Documentation Updates

## Test Plan
Unit tests pass locally:
```
Longest suite times:
  226.9s ripple.tx.NFToken
  126.5s ripple.tx.NFTokenBurn
   66.8s ripple.tx.Offer
   47.9s ripple.app.TheoreticalQuality
   41.6s ripple.tx.NFTokenDir
   38.2s ripple.app.ShardArchiveHandler
   35.5s ripple.app.ValidatorSite
   33.3s ripple.app.AMMExtended
   32.3s ripple.app.AMM
   31.0s ripple.tx.ReducedOffer
989.4s, 211 suites, 1792 cases, 633788 tests total, 0 failures
```

### CODE REVIEW

**Review by seelabs — APPROVED** [review]
:+1: and I agree with the motivation for the change.

File: src/ripple/app/tx/impl/SetAccount.cpp [hunk 1]
```diff
@@ -604,6 +604,8 @@ SetAccount::doApply()
     if (uFlagsIn != uFlagsOut)
         sle->setFieldU32(sfFlags, uFlagsOut);
 
+    ctx_.view().update(sle);
```
[the remark this answers is not in the document]
Comment (mvadari) [reply src/ripple/app/tx/impl/SetAccount.cpp:607]: Yes, seems fine.
```js
"AffectedNodes": [
    {
        "ModifiedNode": {
            "FinalFields": {
                "Account": "rHb9CJAWyB4rj91VRWn96DkukG4bwdtyTh",
                "Balance": "99999999999999980",
                "Flags": 0,
                "OwnerCount": 0,
                "Sequence": 3
            },
            "LedgerEntryType": "AccountRoot",
            "LedgerIndex": "2B6AC232AA4C4BE41BF49D2459FA4A0347E1B543A4C92FCEE0821C0201E2E9A8",
            "PreviousFields": {
                "Balance": "99999999999999990",
                "Sequence": 2
            },
            "PreviousTxnID": "B8789280C43EC42AC5780D6C4C2B97F3A09EE19C6484DF77DC5BF21C33C42CF8",
            "PreviousTxnLgrSeq": 3
        }
    }
],
```

**Comment by intelliot** [comment]
If "this isn't technically a bug", then under "Type of Change", why not call it a refactor?

**Review by ximinez — APPROVED** [review]

**Comment by intelliot** [comment]
@mvadari Can you confirm this is ready to merge? (Or if not, let us know who should re-review)

If ready, I'll put the `Passed` label on it.

**Pull request closed without merging**